src/tiled/variantpropertymanager.cpp

```diff
@@ -126,6 +135,32 @@ QString VariantPropertyManager::valueText(const QtProperty *property) const
         QVariant value = mValues[property].value;
         int typeId = propertyType(property);
 
+        if (typeId == objectIdTypeId()) {
+            ObjectId id = value.value<ObjectId>();
+
+            if (MapDocument *document = DocumentManager::instance()->currentDocument()) {
```

Comment:
How about a `Map*` and an `int` referring to tile id for tile collision objects, or `0` for other objects? Does that uniquely identify all objects to your knowledge?